specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/stable/2022-03-01/fluxconfiguration.json

```diff
@@ -0,0 +1,1115 @@
+          "description": "Name of a local secret on the Kubernetes cluster to use as the authentication secret rather than the managed or user-provided configuration secrets.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "BucketPatchDefinition": {
+      "description": "Parameters to reconcile to the GitRepository source kind type.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration S3 bucket.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "bucketName": {
+          "description": "The bucket name to sync from the url endpoint for the flux configuration.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "insecure": {
+          "description": "Specify whether to use insecure communication when puling data from the S3 bucket.",
+          "type": "boolean",
+          "x-nullable": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster git repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "x-nullable": true
+        },
+        "accessKey": {
+          "description": "Plaintext access key used to securely access the S3 bucket",
+          "type": "string",
+          "x-nullable": true
+        },
+        "localAuthRef": {
+          "description": "Name of a local secret on the Kubernetes cluster to use as the authentication secret rather than the managed or user-provided configuration secrets.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "KustomizationDefinition": {
+      "description": "The Kustomization defining how to reconcile the artifact pulled by the source type on the cluster.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "name": {
+          "description": "Name of the Kustomization, matching the key in the Kustomizations object map.",
+          "type": "string",
+          "readOnly": true
+        },
+        "path": {
+          "description": "The path in the source reference to reconcile on the cluster.",
+          "type": "string",
+          "default": ""
+        },
+        "dependsOn": {
+          "description": "Specifies other Kustomizations that this Kustomization depends on. This Kustomization will not reconcile until all dependencies have completed their reconciliation.",
+          "type": "array",
+          "items": {
+            "description": "Kustomization name of dependency.",
+            "type": "string"
```

Comment:
Previously the type of the item here is `DependsOnDefinition`(see [here](https://github.com/Azure/azure-rest-api-specs/blob/a11e5d03047c44a2a506079292f5b80e4df45fb4/specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2022-01-01-preview/fluxconfiguration.json#L652)), but now it is just a string. This is a breaking change to the SDKs and it breaks backward compatibility because it changes the structure of the json payload as well. Previously, the JSON looks like
```
"dependsOn": [ { "kustomizationName": "something" } ]
```
and now it should look like
```
"dependsOn": [ "something" ]
```
Please double confirm this change, especially from the service code side - and, if possible, please consider revert this.

Reply:
We are coming from a preview version. As far as I am aware, we are allowed to make breaking changes between preview and GA? Is there some additional requirement here?

Reply:
It is allowed. I am just here to double confirm this breaking change with you.

Reply:
Got it, we will be taking this breaking change